libs/beacon/src/beacon_setup_service.cpp

```diff
@@ -713,6 +1004,22 @@ void BeaconSetupService::OnNewSharesPacket(muddle::Packet const &packet,
   OnNewShares(packet.GetSender(), shares);
 }
 
+void BeaconSetupService::OnNewDryRunPacket(muddle::Packet const &packet,
+                                           MuddleAddress const & last_hop)
+{
+  FETCH_UNUSED(last_hop);
+
+  fetch::serializers::MsgPackSerializer serialiser(packet.GetPayload());
+
+  DryRunInfo to_receive;
+  serialiser >> to_receive;
+
+  // TODO(HUT): cabinet check
+  std::lock_guard<std::mutex> lock(mutex_);
+  dry_run_public_keys_[to_receive.public_key]++;
+  dry_run_shares_[packet.GetSender()] = to_receive.sig_share;
```

Comment:
I think the signature shares should be grouped together with the group public key. You should not combine signature shares for different group public keys.

Reply:
Doesn't this mean that someone can corrupt your signing by submitting invalid sig shares? Seems like you could only detect this combinatorially?

In any case, I have put in a fix for this.